### DISCUSSION TITLE
Error when running pet search: "sh: 1: fzf: not found"

### DISCUSSION THREAD

**Discussion by anthony2261** [knqyf263/pet#333] [Q&A] [answered]
As the title says, I seem to need to install the `fzf` package? If it's necessary, it would be good to add this in the docs.

Using WSL2 (ubuntu 20.04). `pet` version 1.0.0

![image](https://github.com/user-attachments/assets/d5a8651e-3504-43e0-a183-19170cf2d3a1)

====================

**Update 1:**

Not really a `pet` issue, but just adding here in case others fall into the same issue:

After installing the fzf package via `sudo apt install fzf`, I get this error: 

![image](https://github.com/user-attachments/assets/64d98331-77b5-4c49-90e1-6729e0136de4)

After doing some digging, it seems the latest version of `fzf` on `ubuntu 20.04` via `apt` is v0.20.0 ([docs](https://github.com/junegunn/fzf?tab=readme-ov-file#linux-packages)), and the `--pointer` argument was added in version `0.21.0` .

But when installing fzf with homebrew, it worked fine 👍

====================

**Update 2:**

The search command doesn't actually execute the command, it just prints it for me

![image](https://github.com/user-attachments/assets/ea717fb6-efc7-441f-accd-adb7e8bf111f)

**Comment by RamiAwar** [comment, marked as answer]
Indeed - we should document this better to prevent confusion.

As for the search command, this is actually expected behavior. It seems a bit weird at first - why doesn't Pet just print the command and set the shell cursor to be at the end of it, allowing the user to execute?

That goes back to how shells and programs work. `pet search` is a command being run. When commands end, the shell moves to a state of waiting for the next command.

The only way of 'controlling' the cursor would be by spawning a subshell, and then printing to stdout inside that. But sub-shells are a spicy can of worms that I'd prefer not to open in this case. 

Also, we recommend installing the shell profile shortcuts as the main way of using Pet. That's also the standard in similar CLI tools to get around this constraint. #288

**Reply by anthony2261** [reply]
Thanks for this!
